packages/maps/src/lib/mapControlLocationSearch/MapGeocoderAdapterMapBox.ts

```diff
@@ -0,0 +1,123 @@
+import { GREATER_LONDON_BOUNDS_PADDED } from '@ldn-viz/maps';
+import type {
+	GeolocationCoords,
+	GeolocationBounds,
+	GeolocationNamed,
+	GeocoderAdapter
+} from '@ldn-viz/ui';
+
+type MapBoxFeature = {
+	id: string;
+	text: string; // name
+	place_name: string; // address
+	center: [number, number];
+	bbox?: [number, number, number, number]; // bounds
+	[otherOptions: string]: unknown;
+};
+
+type MapBoxFeatureCollection = {
+	features: MapBoxFeature[];
+	[otherOptions: string]: unknown;
+};
+
+// MapGeocoderAdapterMapBox adapter uses the MapBox API to source locations.
+//
+// Due to licensing this adapter can only be used within @ldn-viz maps and
+// result locations must be shown on the map.
+export class MapGeocoderAdapterMapBox implements GeocoderAdapter {
+	private _token: string = '';
+	private _resultCount: number = 5;
+
+	constructor(token: string, resultCount = 5) {
+		this._token = token;
+		this.setResultCount(resultCount);
+	}
+
+	// GeocoderAdapter functions.
+
+	search(text: string) {
+		const url = buildUrl(text, this._token, this._resultCount);
+		return fetch(url)
+			.then((res) => res.json())
+			.then(transformGeoJSONToGeolocationNameds);
+	}
+
+	attribution() {
+		return {
+			text: 'Powered by MapBox',
+			link: 'https://docs.mapbox.com/api/search/'
+		};
+	}
+
+	// MapGeocoderAdapterMapBox functions.
+
+	setResultCount(resultCount: number): MapGeocoderAdapterMapBox {
+		if (resultCount > 10) {
+			resultCount = 10;
+			console.warn(
+				`[MapGeocoderAdapterMapBox] maximum MapBox API result count is 10. ${resultCount} is too high so defaulting to 10`
+			);
+		} else if (resultCount < 1) {
+			resultCount = 1;
+			console.warn(
+				`[MapGeocoderAdapterMapBox] minimum MapBox API result count is 1. ${resultCount} is too low so defaulting to 1`
+			);
+		}
+
+		this._resultCount = resultCount;
+		return this;
+	}
+}
+
+const buildUrl = (text: string, token: string, resultCount: number): string => {
+	text = encodeURIComponent(text);
+
+	const queryString = new URLSearchParams({
+		access_token: token,
+		bbox: GREATER_LONDON_BOUNDS_PADDED.flat().toString(),
+		autocomplete: true.toString(),
+		limit: resultCount.toString()
+	});
+
+	return `https://api.mapbox.com/geocoding/v5/mapbox.places/${text}.json?${queryString}`;
+};
+
+const transformGeoJSONToGeolocationNameds = (
+	geojson: MapBoxFeatureCollection
+): GeolocationNamed[] => {
+	return geojson.features.map((loc) => {
+		return {
+			id: loc.id,
+			name: loc.text,
+			address: removeNameFromAddress(loc.place_name, loc.text),
+			// loc.center isn't always the center of the bbox
+			center: calcCorrectCenter(loc.bbox, loc.center),
```

Comment:
This implies that the "correct" center of a feature is the center of the bounding box, which is not in general true.


If a feature's shape is not convex, then the centre of the bounding box might not even be inside the feature: consider a building shaped like the letter "n". 

I'm not sure whether or not adjusting the provided center positions like this is a good idea, but if it is then the function that does this should be called something like `calcBoundingBoxCenter` rather than `calculateCorrectCenter`.

Reply:
Hmmm, when searching for _'Canning Town'_ the symbol point is placed quite far away from the BBOX center which confused me (as a user). I know boundaries can be quite irregular.

I suppose it depends on the level of accuracy a user needs when location searching? As a user, you may want to pan around then come back to your searched spot. A centered symbol would be the better reference point.

In a future update we could put a border on the map representing the BBOX instead? Might look a bit ugly.

I'll rename the function to `calcBoundingBoxCenter`.

Reply:
Or just not have a marker for BBOX locations?